### PR TITLE
feat(apigateway): support vpc endpoints

### PR DESCRIPTION
**Pull request by mb-dev**
VPC Endpoint ID allow invoking private api gateway using the endpoint dns name. Cloudformation recently added support for specifying VPC Endpoint IDs in API Gateway configuration:
https://aws.amazon.com/about-aws/whats-new/2019/12/aws-cloudformation-updates-for-api-gateway-codepipeline-s3-iam-ecs-rds-es-lambda-and-more/

More information:
https://docs.aws.amazon.com/apigateway/latest/developerguide/associate-private-api-with-vpc-endpoint.html

The vpc endpoints ids are a property of `endpointConfiguration`, similar to the existing `endpointTypes`. So I added the new property in a similar way.

Note: This is my first PR for aws-cdk, let me know if any changes are needed.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigateway/lib/restapi.ts [hunk 1]
```diff
@@ -116,6 +116,14 @@ export interface RestApiProps extends ResourceOptions {
    */
   readonly endpointTypes?: EndpointType[];
 
+  /**
+   * A list of the VPC Endpoint IDs of the API. Use this property when creating
+   * a private API.
+   *
+   * @default - No VPC Endpoint IDs.
+   */
+  readonly vpcEndpointIds?: string[];
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/restapi.ts]: I believe this should be of type `IVpcEndpoint` from the `@aws-cdk/aws-ec2` module.

File: packages/@aws-cdk/aws-apigateway/lib/restapi.ts [hunk 2]
```diff
@@ -116,6 +116,14 @@ export interface RestApiProps extends ResourceOptions {
    */
   readonly endpointTypes?: EndpointType[];
 
+  /**
+   * A list of the VPC Endpoint IDs of the API. Use this property when creating
+   * a private API.
+   *
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/restapi.ts]: A couple of sentences of what behaviour this property enables is always useful. 

A good technique is to transcribe a couple of sentences from [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/associate-private-api-with-vpc-endpoint.html) and then link to it with a `@see` tag.

File: packages/@aws-cdk/aws-apigateway/lib/restapi.ts (hunk 1 above)
Comment (mb-dev) [reply packages/@aws-cdk/aws-apigateway/lib/restapi.ts]: Updated to use VPCEndpoint. This adds dependency between api gateway and ec2. I moved the dependency from dev dependency to a real dependency with peer requirement.

File: packages/@aws-cdk/aws-apigateway/lib/restapi.ts (hunk 2 above)
Comment (mb-dev) [reply packages/@aws-cdk/aws-apigateway/lib/restapi.ts]: I added the link and brief explanation that it creates DNS alias.

**Comment by mb-dev** [comment]
@nija-at I added validation that endpoint types include private

**Comment by nija-at** [comment]
Closing this PR due to inactivity. Feel free to re-open it when you're ready.

**Comment by stefanolczak** [comment]
It looks like this is resolved by #6078

**Pull request closed without merging**